src/app/admin/components/DataTableSectionUser.tsx

```diff
@@ -0,0 +1,356 @@
+  "w-full rounded-md border border-awesomer-purple bg-white p-2 focus:outline-none focus:ring-1 focus:ring-awesomer-purple";
+const CHANGE_PAGE_BUTTON_STYLING =
+  "rounded-md border border-awesomer-purple bg-white px-6 hover:bg-awesomer-purple hover:text-white";
+const CHANGE_PAGE_BUTTON_TEXT_STYLING = "rounded-md bg-white p-2 px-8";
+
+const search_icon = "/svgs/admin/search_icon.svg";
+
+const entries_per_page = 10;
+
+interface DataTableProps {
+  tableHeaders: Array<{ columnHeader: string }>;
+  userData?: Array<Partial<Schema["User"]["type"]>>;
+  tableDataMutation: any;
+}
+
+const DataTableSectionUser = (props: DataTableProps) => {
+  const { control, handleSubmit } = useForm<Schema["User"]["type"]>();
+
+  const onSubmit: SubmitHandler<Schema["User"]["type"]> = (data) => {
+    const strippedObject = userData.find((user) => user.email === data.email);
+    console.log(data);
+    if (strippedObject) {
+      data.id = strippedObject.id ?? "";
+      tableDataMutation.mutate(data);
+      console.log(data);
+    } else {
+      console.error("User not found");
+      toast.error("❌ User cannot be found", {
+        position: "top-right",
+        autoClose: 5000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        progress: 0,
+        theme: "light",
+        transition: Bounce,
+      });
+    }
+  };
+
+  const { tableHeaders, userData = [], tableDataMutation } = props;
+
+  const [editModes, setEditModes] = useState(
+    Array(userData.length).fill(false),
+  );
+
+  const [showDeletePopup, setShowDeletePopup] = useState(false);
+  const [recordToDeleteId, setRecordToDeleteId] = useState(
+    "" as string | undefined,
+  );
+  const [searchQuery, setSearchQuery] = useState("");
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [currentPageData, setCurrentPageData] =
+    useState<Array<Partial<Schema["User"]["type"]>>>(userData);
+
+  const filteredData = useMemo(() => {
+    return userData.filter((rowData) =>
+      Object.values(rowData).some(
+        (value) =>
+          value !== null &&
+          value !== undefined &&
+          value.toString().toLowerCase().includes(searchQuery.toLowerCase()),
+      ),
+    );
+  }, [userData, searchQuery]);
+  const startIndex = (currentPage - 1) * entries_per_page;
+  const endIndex = Math.min(startIndex + entries_per_page, filteredData.length);
+
+  useEffect(() => {
+    const totalPages = Math.ceil(filteredData.length / entries_per_page);
+    const currentPageData = filteredData
+      .slice(startIndex, endIndex)
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .map(({ id, ...allOtherFields }) => allOtherFields);
+
+    setTotalPages(totalPages);
+    setCurrentPageData(currentPageData);
+  }, [filteredData, currentPage, entries_per_page]);
+
+  const handlePreviousPage = () => {
+    setCurrentPage((prevPage) => Math.max(prevPage - 1, 1));
+  };
+
```

Comment:
Remove ()

Reply:
for some reason when deleting there is a "Too many re-renders" react error so I will leave it for now